### PR TITLE
Cameras sample

### PR DESCRIPTION
**Pull request by ben-clayton**
Implements an arcball and WASD-style camera sample.

Requested in https://github.com/webgpu/webgpu-samples/discussions/291.

### CODE REVIEW

File: src/sample/cameras/main.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    renderPassDescriptor.colorAttachments[0].view = context
+      .getCurrentTexture()
+      .createView();
+
+    const commandEncoder = device.createCommandEncoder();
+    const passEncoder = commandEncoder.beginRenderPass(renderPassDescriptor);
+    passEncoder.setPipeline(pipeline);
+    passEncoder.setBindGroup(0, uniformBindGroup);
+    passEncoder.setVertexBuffer(0, verticesBuffer);
+    passEncoder.draw(cubeVertexCount, 1, 0, 0);
```
Comment (Kangz) [inline src/sample/cameras/main.ts:241]: u-nit: the last three arguments to draw are optional!

**Review by Kangz — APPROVED** [review]

File: src/sample/cameras/camera.ts [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    // Calculate the new target velocity
+    const digital = input.digital;
+    const deltaRight = sign(digital.right, digital.left);
+    const deltaUp = sign(digital.up, digital.down);
+    const targetVelocity = vec3.create();
+    const deltaBack = sign(digital.backward, digital.forward);
+    vec3.addScaled(targetVelocity, this.right, deltaRight, targetVelocity);
+    vec3.addScaled(targetVelocity, this.up, deltaUp, targetVelocity);
+    vec3.addScaled(targetVelocity, this.back, deltaBack, targetVelocity);
+    vec3.mulScalar(targetVelocity, this.movementSpeed, targetVelocity);
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/sample/cameras/camera.ts:188]: Just like a proper FPS!

[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/sample/cameras/camera.ts:189]: Done

File: src/sample/cameras/input.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  const analog = {
+    x: 0,
+    y: 0,
+    zoom: 0,
+  };
+  let mouseDown = false;
+
+  const setDigital = (e: KeyboardEvent, value: boolean) => {
+    switch (e.key) {
+      case 'w':
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/sample/cameras/input.ts]: Done

File: src/sample/cameras/input.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      case 'D':
+        digital.right = value;
+        e.preventDefault();
+        e.stopPropagation();
+        break;
+      case ' ':
+        digital.up = value;
+        e.preventDefault();
+        e.stopPropagation();
+        break;
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/sample/cameras/input.ts:73]: Done

File: src/sample/cameras/main.ts (hunk 1 above)
Comment (ben-clayton) [reply src/sample/cameras/main.ts]: Done, and: https://github.com/webgpu/webgpu-samples/pull/296

**Pull request merged**